packages/java-parser/src/productions/blocks-and-statements.js

```diff
@@ -57,10 +61,13 @@ function defineRules($, t) {
 
   // https://docs.oracle.com/javase/specs/jls/se11/html/jls-14.html#jls-LocalVariableType
   $.RULE("localVariableType", () => {
-    $.OR([
-      { ALT: () => $.SUBRULE($.unannType) },
-      { ALT: () => $.CONSUME(t.Var) }
-    ]);
+    $.OR({
+      DEF: [
+        { ALT: () => $.SUBRULE($.unannType) },
+        { ALT: () => $.CONSUME(t.Var) }
+      ],
+      IGNORE_AMBIGUITIES: true
```

Comment:
Not sure how to enforce not using 'var' in unannType

Reply:
can var be used as an identifier? is that the cause of the ambiguity?

Reply:
If that is the case (I have some fuzzy memory of limitations for the `var` keyword.
You may be able to define a IdentifierButNotVar Token using token categories.

Reply:
Yes, that's right. I think I will try to improve this in a following PR, adding a token and change some parser logic seems out of the scope of this PR

Reply:
Yes definitively out of scope